tests/unit/test_unmarshal.py

```diff
@@ -0,0 +1,264 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+
+from craft_parts import unmarshal
+
+
+@pytest.mark.parametrize(
+    "data,value,error",
+    [
+        [{}, "", None],
+        [{"test": ""}, "", None],
+        [{"test": "foo"}, "foo", None],
+        [{"test": False}, "", "'test' must be a string"],
+    ],
+)
+def test_unmarshal_string(data, value, error):
+    udata = unmarshal.DataUnmarshaler(data)
```

Comment:
I'd suggest breaking out the error scenarios into its own test(s) to reduce test complexity.

Reply:
Reorganize tests to make them easier to read.